paperqa/docs.py

```diff
@@ -328,9 +333,7 @@ async def aadd(  # noqa: PLR0912
             except (json.JSONDecodeError, AttributeError):
                 # json.JSONDecodeError: clean_text was not actually JSON
                 # AttributeError: citation_json was not a dict (e.g. a list)
-                # NOTE: we only want to warning log when we fail on title and
-                # authors we may not care if we fail on DOI here
```

Comment:
Why delete this comment? It adds value imo, DOI extraction is the only critical one

Reply:
It's a restatement of what's in the logger warning imo

Reply:
So the thinking here is extracting DOI is really valuable, extracting title and authors is a nice-to-have

Why do you think it's a restatement? The message below doesn't talk about DOI being more important than title and authors

Reply:
So I decided this comment can be removed, as it's suboptimal.

Basically:
- Succeeding in an extraction here saves us from trying to infer this information in downstream logic
- We get different gains from extracting title, author, and DOI

This comment was trying to make one aware of that, but clearly it's not doing its job as an informative comment.

It's unclear to me if a suboptimal comment making one aware of some behavior is better or worse than no comment

Reply:
Oh - yea I guess you're right. Then I deleted the comment because it doesn't reflect the code which previously just threw an exception with any failure. Maybe @mskarlin -  can you comment on if the comment was correct or the code was correct?

Reply:
Fwiw it didn't throw an exception earlier, it just logged a stack trace and continued. The user didn't actually experience a crash, just the code took a different path and inferred the title/author/DOI later

Reply:
Oh - I thought `logger.exception` actually created an exception. Thanks - didn't realize that.